src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Syntax/Names.cs

```diff
@@ -4,23 +4,15 @@
 
 namespace Microsoft.CSharp.RuntimeBinder.Syntax
 {
-    internal class Name
+    internal sealed class Name
```

Comment:
With just one reference-type field this could probably be either made a struct or just have `string` used directly, but there'd have to be some changes around comparison with null and/or identity-based hashing (if replacing with string entirely) for that not to be completely trivial, so won't try that here.